pxr/imaging/plugin/hioOiio/oiioImage.cpp

```diff
@@ -713,13 +713,27 @@ HioOIIO_Image::ReadCropped(int const cropTop,
     // If needed, convert double precision images to float
     bool res = false;
     if (imageInput->spec().format == TypeDesc::DOUBLE) {
-        res = imageInput->read_image(TypeDesc::FLOAT,
+        res = imageInput->read_image(
+#if OIIO_VERSION >= 30000
```

Comment:
We could probably push this farther back in OIIO versions if we wanted to, as the overload that was being used was implemented this way fairly early in the 2.x series as far as I can tell. For now I opted to restrict it to just handling the build breakage, but let me know if I should try to trace back to an earlier version.

Reply:
@mattyjams in https://github.com/PixarAnimationStudios/OpenUSD/pull/3365/files @jessey-git has the same-ish changes. I notice you have got imageInput->current*, and @jessey-git has `_subimage`, and `_miplevel` instead. @jessey-git hasn't ifdef'd this block. so I'm wondering if we can reconcile these two PRs

1. should it be `imageInput->` or `_whatever`?
2. can we still support 2.x and remove the ifdef's?

thanks!

Reply:
Heya, the reason I decided to use `_subimage, _miplevel` is that we already use those when making the `seek_subimage` check on line 691 of the same file.  Whichever way we go, can we make both call sites match (I think they should match)?

Reply:
Actually, this way is probably fine really. You can disregard my post above :)